src/status_im2/contexts/chat/messages/pin/events.cljs

```diff
@@ -33,34 +33,46 @@
              (assoc-in [:pagination-info chat-id :all-pin-loaded?]
                        (empty? cursor)))}))
 
-(rf/defn receive-signal
-  [{:keys [db]} pin-messages]
-  (let [{:keys [chat-id]} (first pin-messages)]
-    (when (= chat-id (db :current-chat-id))
-      (let [{:keys [chat-id]}           (first pin-messages)
-            already-loaded-pin-messages (get-in db [:pin-messages chat-id] {})
-            already-loaded-messages     (get-in db [:messages chat-id] {})
-            all-messages                (reduce (fn [acc {:keys [message_id pinned from]}]
-                                                  ;; Add to or remove from pinned message list, and
-                                                  ;; normalizing pinned-by property
-                                                  (let [current-message     (get already-loaded-messages
-                                                                                 message_id)
-                                                        current-message-pin (merge current-message
-                                                                                   {:pinned    pinned
-                                                                                    :pinned-by from})]
-                                                    (cond-> acc
-                                                      (nil? pinned)
-                                                      (dissoc message_id)
+(defn remove-pinned-message
+  [db pinned-message-js]
```

Comment:
It's more idiomatic to not name the host data with the suffix `-js`, or `-java`, etc. Adding a type hint should be enough, and also the dev will imply the data type by the usage of `oops`, which we prefer to use instead of direct interop macros (according to the guidelines).